pkg/operator/encryption/controllers/migration_controller.go

```
@@ -216,25 +216,40 @@ func (c *migrationController) migrateKeysIfNeededAndRevisionStable(ctx context.C
 
 		// idem-potent migration start
-		finished, result, when, err := c.migrator.EnsureMigration(gr, grActualKeys.WriteKey.Key.Name)
+		finished, result, when, err := c.migrator.EnsureMigration(gr, migrationWriteKey)
 		if err == nil && finished && result != nil && time.Since(when) > migrationRetryDuration {
 			// last migration error is far enough ago. Prune and retry.
 			if err := c.migrator.PruneMigration(gr); err != nil {
 				errs = append(errs, err)
 				continue
 			}
-			finished, result, when, err = c.migrator.EnsureMigration(gr, grActualKeys.WriteKey.Key.Name)
+			finished, result, when, err = c.migrator.EnsureMigration(gr, migrationWriteKey)
```

golangci-lint (2.13.2)

[error] ineffectual assignment to when

(ineffassign)